File: java/src/main/java/io/fivexer/sdk/model/WorkspaceStats.java
package io.fivexer.sdk.model;

/** Result of {@code GET /v1/stats}: workspace counts, worker total, and the monthly meter. */
public class WorkspaceStats {
    private String plan;
    private java.util.Map<String, Integer> tasks;
    private int workers;
    private Meter meter;
    private QueueStats queue;
    private MatchingPolicy matching;

    public String getPlan() { return plan; }
    public java.util.Map<String, Integer> getTasks() { return tasks; }
    public int getWorkers() { return workers; }
    public Meter getMeter() { return meter; }

    /** Queue depth and per-worker load. Absent on data-plane-only deployments. */
    public QueueStats getQueue() { return queue; }

    /** The live balancer policy applied to bulk matching passes. */
    public MatchingPolicy getMatching() { return matching; }

    public static class Meter {
        private String period;
        private int matchedTasks;
        private int includedTasksPerMonth;

        public String getPeriod() { return period; }
        public int getMatchedTasks() { return matchedTasks; }
        public int getIncludedTasksPerMonth() { return includedTasksPerMonth; }
    }
}

File: php/src/Model/WorkspaceStats.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/** Result of GET /v1/stats: workspace counts, worker total, and the monthly meter. */
final class WorkspaceStats
{
    /**
     * @param array<string, int> $tasks
     */
    public function __construct(
        public readonly string $plan,
        public readonly array $tasks,
        public readonly int $workers,
        public readonly string $meterPeriod,
        public readonly int $meterMatchedTasks,
        public readonly int $meterIncludedTasksPerMonth,
        /** Queue depth and per-worker load; null on data-plane-only deployments */
        public readonly ?QueueStats $queue = null,
        /** The live balancer policy applied to bulk matching passes */
        public readonly ?MatchingPolicy $matching = null,
    ) {
    }

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        $meter = $data['meter'] ?? [];
        /** @var array<string, int> $tasks */
        $tasks = $data['tasks'] ?? [];
        return new self(
            plan: (string) ($data['plan'] ?? ''),
            tasks: \array_map('intval', $tasks),
            workers: (int) ($data['workers'] ?? 0),
            meterPeriod: (string) ($meter['period'] ?? ''),
            meterMatchedTasks: (int) ($meter['matchedTasks'] ?? 0),
            meterIncludedTasksPerMonth: (int) ($meter['includedTasksPerMonth'] ?? 0),
            queue: isset($data['queue']) ? QueueStats::fromArray($data['queue']) : null,
            matching: isset($data['matching']) ? MatchingPolicy::fromArray($data['matching']) : null,
        );
    }
}
